bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java

```diff
@@ -0,0 +1,999 @@
+			Point expectedSize) {
+		IEclipsePreferences preferences = InstanceScope.INSTANCE.getNode("org.eclipse.ui.editors"); //$NON-NLS-1$
+		boolean placeAtBottomOfPage = preferences.getBoolean("findreplaceoverlayatbottom", false); //$NON-NLS-1$
+		Point scrollBarSize = ((Scrollable) targetTextWidget).getVerticalBar().getSize();
+
+		int newX = targetOrigin.x + targetBounds.width - expectedSize.x - scrollBarSize.x
+				- ((StyledText) targetTextWidget).getRightMargin();
+		int newY = targetOrigin.y;
+		if (placeAtBottomOfPage) {
+			newY += targetBounds.height - expectedSize.y;
+		}
+		return new Point(newX, newY);
+	}
+
+	/**
+	 * When making the text-bar 100% small and then regrowing it, we want the text
+	 * to start at the first character again.
+	 */
+	private void repositionTextSelection() {
+		if (okayToUse(searchBar) && !searchBar.isFocusControl()) {
+			searchBar.setSelection(0, 0);
+		}
+		if (okayToUse(replaceBar) && !replaceBar.isFocusControl()) {
+			replaceBar.setSelection(0, 0);
+		}
+	}
+
+	private void positionToPart() {
+		getShell().requestLayout();
+		if (!(targetPart instanceof StatusTextEditor)) {
+			return;
+		}
+
+		StatusTextEditor textEditor = (StatusTextEditor) targetPart;
+		Control targetWidget = textEditor.getSourceViewer().getTextWidget();
+		if (targetWidget == null || targetWidget.isDisposed()) {
+			this.close();
+			return;
+		}
+
+		Point targetOrigin = targetWidget.toDisplay(0, 0);
+		Rectangle targetBounds = targetWidget.getBounds();
+
+		int newWidth = getIdealDialogWidth(targetBounds);
+		int newHeight = container.computeSize(SWT.DEFAULT, SWT.DEFAULT).y;
+
+		Point newPosition = getNewPosition(targetWidget, targetOrigin, targetBounds, new Point(newWidth, newHeight));
+
+		getShell().setSize(new Point(newWidth, newHeight));
+		getShell().setLocation(newPosition);
+		getShell().layout(true);
+
+		repositionTextSelection();
+	}
+
+	private String getFindString() {
+		return searchBar.getText();
+	}
+
+	private String getReplaceString() {
+		if (replaceBar.isDisposed()) {
+			return ""; //$NON-NLS-1$
+		}
+		return replaceBar.getText();
+
+	}
+
+	private void performSingleReplace() {
+		findReplaceLogic.performReplaceAndFind(getFindString(), getReplaceString());
+		evaluateFindReplacerStatus();
+	}
+
+	private void initFindStringFromSelection() {
+		String initText = findReplaceLogic.getTarget().getSelectionText();
+		if (initText.isEmpty()) {
+			return;
+		}
+		if (initText.contains("\n")) { //$NON-NLS-1$
+			findReplaceLogic.deactivate(SearchOptions.GLOBAL);
+			searchInSelectionButton.setSelection(true);
+		} else {
+			searchBar.setText(initText);
+			searchBar.setSelection(0, initText.length());
+		}
+	}
+
+	private void evaluateFindReplacerStatus() {
+		Color warningColor = JFaceColors.getErrorText(getShell().getDisplay());
+		IFindReplaceStatus status = findReplaceLogic.getStatus();
+
+		if (status instanceof InvalidRegExStatus || findReplacerHasNoMatchStatus(status)) {
```

Comment:
The reason we are doing this is not because we want to see whether the status was somehow "unsuccessful" but rather that the old FindReplaceDialog had very specific types of responses to very specific types of statuses.
Now, that we have largely moved away from the beeping-behavior, I believe it is acceptable to think about new behavior. I will use "!status.isInputValid()" as condition for showing the red error-font.

Reply:
The issue with my "solution" is the following:

StatusCode.WRAP will result in an "unsuccessful" state
StatusCode.NO_MATCH will result in a state with valid input

We only want to display an error if NO_MATCH is true but NOT if the search was wrapped (this is be particularly annoying if there is only one single item that matches the incremental search and our search-text turns red even though there IS a match).

I will reinstate the old behavior until we find a better solution